### ISSUE TITLE
Feature Request: Multiple key support

### ISSUE THREAD

**Issue by mikebryant** [bitnami-labs/sealed-secrets#120]
We have many kubernetes clusters. For lots of secrets, we encrypt them per-cluster, as they're unique to that cluster.

For some secrets, we would like to encrypt them once, and have all clusters be able to decrypt those secrets. Examples might be, the newrelic or pagerduty API keys.
Each cluster would have the dynamically generated key that's unique to it, and we would like to use that key to encrypt the global decryption key.

This could be modelled by allowing SealedSecrets to optionally reference a different decryption key, potentially?

**Comment by mellena1** [comment]
I'd like to throw another use case in here too that I think would work with the suggested solution as well.

We would like to have a key per namespace basically. Our projects are planned to be separated by namespace, and we don't want every project to have the same key (as to not have people on one project be able to grab secrets from another). So I think a label selector would work well here for us. Could put a key in each namespace and label it based on that.

Any thoughts?

**Comment by scottrigby** [comment]
I was about to make an issue from the chat at #TGIK8S at kubecon - but this is already here 👍 Also cross referencing https://github.com/bitnami-labs/sealed-secrets/issues/101#issuecomment-389380289. Looking fwd to checking this out more soon!

**Comment by johnraz** [comment]
@mellena1 Isn't this usecase already covered today as the namespace used to generate the initial secret resource(the cleartext one) is used at encryption time to make sure that the secret can't be decrypted to another namespace by the controller later on.  /cc @anguslees  to be sure I'm not mistaken here 😉

**Comment by mellena1** [comment]
@johnraz Yeah I think you are correct in that. I realized that after I made that post. So more info on how that works would be cool!

**Comment by anguslees** [comment]
@mellena1 right, your case above is covered by the existing namespace verification when decrypting.  See everywhere that mentions "namespace" in the README (in Usage and Details sections) ;)

I would be interested in where you (and the 3 +1's) wanted to find more information on this, because the existing README isn't working very well - this is the #1 most misunderstood aspect of sealed-secrets.  If you have suggestions, perhaps reply in a new issue so we don't further pollute this one... (and thanks in advance)

**Comment by johnraz** [comment]
I created #136 to address the documentation issue.

**Comment by aslafy-z** [comment]
Issue is not stale
